CMake/Findre2.cmake

```diff
@@ -0,0 +1,29 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+find_library(re2_lib re2)
+find_path(re2_include re2/re2.h)
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(re2 REQUIRED_VARS re2_lib re2_include)
+
+if(re2_FOUND)
+  set(re2_LIBRARIES ${re2_lib})
+  set(re2_INCLUDE_DIRS ${re2_include})
+
+  add_library(re2::re2 UNKNOWN IMPORTED)
```

Comment:
Maybe basic question but why do we need to do this ? wont find_package above do this ?

Reply:
No, `find_library` will not create any targets, that's why I added this file in the first place. `find_package_handle_standard_args` only parses the arguments and sets `re2_FOUND`/emits errors if the lib was not found.